### PR TITLE
[AD-201] Write TUI usage guide

### PR DESCRIPTION
**Pull request by maksbotan**

### CODE REVIEW

File: docs/usage-tui.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
+# Ariadne TUI user guide
+
+This is a user guid for Ariadne text-based user interface (TUI), which runs in any terminal
```
Comment (gromakovsky) [inline docs/usage-tui.md]: s/guid/guide

File: docs/usage-tui.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+## About mouse support
+
+Note that almost any action accomplishable with keyboard can also be performed with a mouse in an
+intuitive way. This guide will mostly focus on keyboard shortcuts, mentioning mouse where its usage
+is not obvious.
+
+Also please note that while Ariadne (yet) has no native support for text selection with mouse, you
+can still select text if you need, by using your terminal emulator's features. To select text on
+Linux you usually need to hold `Shift` key and on macOS -- the `Fn` key.
```
Comment (gromakovsky) [inline docs/usage-tui.md]: For me it's `Alt`, not `Fn`, so something less concrete should be here.

File: docs/usage-tui.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+On the `Wallet` tab you can see all your wallets with their accounts and addresses organized in a
+tree. Right of the tree you can see details on the selected wallet, account or address.
+
+To create a new wallet, select `[ + Add wallet ]` line in the tree, so that a special widget appears
+to the right. There you can either create a completely new wallet, by specifying its name and a
+passphrase, or restore an existing one. When a new wallet is created, Ariadne will print you its
+mnemonic. **Please save it somewhere safe**.
+
+Note that Ariadne mnemonics consist of 13 words, as opposed to 12-words mnemonics of Daedalus. This
```
Comment (gromakovsky) [inline docs/usage-tui.md]: Note necessary 13, depends on `entropy-size`. Can be 13, 17, 21, 25…

File: docs/usage-tui.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    Daedalus's secret file. `full` argument has the same effect as for `restore` command. Please
+    note that the restored wallet will have the same passphrase as in Daedalus.
+
+There is also a command to send a transaction from you wallet: `send`. Since one transaction can
+have multiple outputs, you have to pass these outputs to the `send` command. An output is
+constructed with `tx-out` command, which takes a receiver's address and an amount of ADA you want to
+send to this address. There are several ways to specify the amount of coins:
+
+- Just a number, like `2` or `0.5`. It will be treated as an amount of ADA coins. Note that you
+    can't have a fraction of ADA less than 0.000001 (10^-6).
```
Comment (gromakovsky) [inline docs/usage-tui.md]: Please make `10^-6` look cool (`10⁻⁶`).

**Comment by maksbotan** [comment]
I've described some literals. Do we really need file paths though? There are no useful commands that accept them AFAIU. Also, should I describe hashes (which start with `#`)? I don't get what's their format.

File: docs/usage-tui.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+## About mouse support
+
+Note that almost any action accomplishable with keyboard can also be performed with a mouse in an
+intuitive way. This guide will mostly focus on keyboard shortcuts, mentioning mouse where its usage
+is not obvious.
+
+Also please note that while Ariadne has no native support for text selection with mouse (yet), you
+can still select text by using your terminal emulator's features. On Linux you usually need to hold
+`Shift` while selecting. On macOS please consult your terminal emulator's documentation.
```
Comment (gromakovsky) [inline docs/usage-tui.md]: I'd write `On macOS try holding Alt or consult your terminal emulator's documentation.`, because most people use iTerm, AFAIK, and there Alt worked for me.

File: docs/usage-tui.md [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+constructed with `tx-out` command, which takes a receiver's address and an amount of ADA you want to
+send to this address. There are several ways to specify the amount of coins:
+
+- Just a number, like `2` or `0.5`. It will be treated as an amount of ADA coins. Note that you
+    can't have a fraction of ADA less than 0.000001 (10^¯⁶).
+- A number with `ADA` suffix: `2ADA`, `0.5ADA`. This option is identical to the previous one.
+- A number with `Lovelace` suffix: `100Lovelace`, `500000Lovelace`. One Lovelace is 0.000001 (10^¯⁶)
+    of one ADA. Lovelaces can't be fractional.
+
+For you convenience, `ADA` and `Lovelace` suffixes are case-insensitive.
```
Comment (gromakovsky) [inline docs/usage-tui.md]: s/you/your

File: docs/usage-tui.md [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+- `restore pass: String? name: String? mnemonic: String full: Bool`: restore a wallet from mnemonic.
+    A passphrase is used to encrypt the restored wallet in Ariadne, it does not have to be the same
+    as the passphrase the old wallet was encrypted with. `full` argument specifies whether Ariadne
+    should perform a full restore: find all used accounts and addresses of the restored wallet in
+    the blockchain.
+- `restore-from-daedalus-file name: String? file: FilePath full: Bool`: restore wallet from
+    Daedalus's secret file. `full` argument has the same effect as for `restore` command. Please
+    note that the restored wallet will have the same passphrase as in Daedalus.
+
+There is also a command to send a transaction from you wallet: `send`. Since one transaction can
```
Comment (gromakovsky) [inline docs/usage-tui.md]: Why is `send` so special that you described it separately and didn't provide list of its argument in common notation used above? :thinking:

Comment (gromakovsky) [reply docs/usage-tui.md]: s/you/your

File: docs/usage-tui.md [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+account using `new-account` command &mdash; it will use the currently selected one. There is also a
+`copy-selection` command, which will copy the name of selected item (wallet, account or address) to
+the system clipboard.
+
+### Cardano-related commands
+
+Finally, to the commands which will help you manage your `ADA` wallet. Each command will be
+annotated with its arguments and their types. Optional arguments will be marked with "`?`".
+
+- `new-wallet pass: String? name: String? entropy-size: Int?`: create a new wallet with a
```
Comment (gromakovsky) [inline docs/usage-tui.md]: I'd mention that we use the approach from BIP-39.

File: docs/usage-tui.md [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+    passphrase, name and size of entropy. Size of entropy influences security of your wallet and the
+    length of the mnemonic. Default value is 16 bytes, leading to mnemonics of 13 words.
+- `new-account wallet: String or Word name: String?`: create a new account in the specified or
+    the currently selected wallet. You can specify wallet either by its name or by its (zero-based)
+    index in the wallet tree.
+- `new-address wallet: String or Word account: String or Word pass: String?`: create a new address
+    with an optional passphrase. This command will use either the currently selected account or the
```
Comment (gromakovsky) [inline docs/usage-tui.md]: It makes an impression that the passphrase will be used to encrypt the address, but it's used to decrypt the wallet key. I'd rephrase it somehow. Passphrase must match one used for the wallet creation.

File: docs/usage-tui.md (hunk 7 above)
Comment (maksbotan) [reply docs/usage-tui.md]: Because it's more complicated than a simple command with clear signature. Refactored it a bit.

File: docs/usage-tui.md (hunk 8 above)
Comment (maksbotan) [reply docs/usage-tui.md]: Ok.

File: docs/usage-tui.md (hunk 9 above)
Comment (maksbotan) [reply docs/usage-tui.md]: I understood it the wrong way, so I wrote it the wrong way :) Changed.

File: docs/usage-tui.md [hunk 10]
```diff
@@ -162,18 +174,22 @@ annotated with its arguments and their types. Optional arguments will be marked
...
+command. An output is constructed with `tx-out` command, which takes a receiver's address and an
+amount of ADA you want to send to this address. There are several ways to specify the amount of
+coins:
 
 - Just a number, like `2` or `0.5`. It will be treated as an amount of ADA coins. Note that you
-    can't have a fraction of ADA less than 0.000001 (10^-6).
+    can't have a fraction of ADA less than 0.000001 (10^¯⁶).
 - A number with `ADA` suffix: `2ADA`, `0.5ADA`. This option is identical to the previous one.
-- A number with `Lovelace` suffix: `100Lovelace`, `500000Lovelace`. One Lovelace is 0.000001 (10^-6)
+- A number with `Lovelace` suffix: `100Lovelace`, `500000Lovelace`. One Lovelace is 0.000001 (10^¯⁶)
```
Comment (gromakovsky) [inline docs/usage-tui.md]: This `10^¯⁶` doesn't make sense (should be `10¯⁶`).
